tests/unit/lms/services/canvas_api_test.py

```diff
@@ -1,643 +1,482 @@
+import json
```

Comment:
I'd really recommend not reading this as a diff:

 * The file has been rewritten again from scratch 
 * It covers all of the code where as previously it was only 1/4
 * It's still somehow 200 lines shorter